nengo/config.py

```diff
@@ -84,6 +84,10 @@ def __repr__(self):
         return "<%s[%s]{%s}>" % (type(self).__name__,
                                  self._configures.__name__, ", ".join(params))
 
+    def __setstate__(self, state):
+        for k, v in state.items():
+            setattr(self, k, v)
```

Comment:
I like having `__getattr__`, `__setattr__`, and `__delattr__` all respond to keys starting with underscores in a special manner. It's clearer for me when I look at all three functions and see they have the same general form. So I'd make that change for sure. Would it make sense to do that and implement `__slots__`? I haven't seen the `__slots__` implementation, so I can't really say what I think of it.